Visualisation

jbathmann opened the pull request:
Visualisation and no visualisation added using matplotlib module

zaphod-bx commented:
Cool!

mcwimm commented on ProjectLib/Project.py:
```diff
@@ -56,6 +58,13 @@ def iniTreeTimeLoop(self):
 
+    def iniVisualization(self):
+        arg = self.args["visualization"]
+        self.visualization = (Visualization(arg))
```
useless parenthesis

jbathmann replied:
removed

mcwimm commented on VisualizationLib/NoVis/NoVis.py:
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python3
```
I'm not sure if we really need an extra class for "no visualization". Coudn't we just say if there's no visualization element/ tag in the project file we don't initialize a visualization?

jbathmann replied:
I do not completely agree. By making the tag <visualization> required, the project file becomes more structured. In the end, I'd like to have the required tags:
-tree_dynamics
--AC
---type
--BC
---type
--GADD
---type
-time_loop
--type
-visualization
--type
-output
--type
-tree_population
--type

This gives the user an idea of all functionalities, which have to be considered. In this context, we could also rename our SimpleTest scenarios to sth like NoCompetition/NoGrowth (since they are not modyfying anything).

Of course, this is just a suggestion. What do you, @zaphod-bx  and @waltherm think of this?

mcwimm replied:
I agree with the fixed structure of the xml file, but we could use "none" as text option for visualization.

jbathmann replied:
done

> I agree with the fixed structure of the xml file, but we could use "none" as text option for visualization.

I agree. Just pushed changed tag names

mcwimm replied:
now we still have a class for "no visualization", but without meaning. That's not the idea of classes, is it?

jbathmann replied:
Sorry, I don't get what you mean.
I added a this "NONE" class. It is containing dummy functions. Using this approach the functions in treedynamictimestepping within the timelooplib can stay the same, when we change between NONE (visualisation) and the python visualization.
So there is a meaning (or a use) for the class... Or do you mean something else?

zaphod-bx replied:
I agree with jbathmann, aber mirs wurscht, ob dit NONE oder no visualisation heeßt.

mcwimm commented on VisualizationLib/NoVis/__init__.py:
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python3
```
see NoVis.py

jbathmann replied:
s.o.

zaphod-bx replied:
s.o.

mcwimm commented on VisualizationLib/Visualization.py:
```diff
@@ -0,0 +1,36 @@
+        from .SimplePyplot import SimplePyplot
+        self.visualization = SimplePyplot(args)
+
+    def iniNoVis(self, args):
```
see NoVis.py

jbathmann replied:
s.o.

zaphod-bx replied:
o.s.

mcwimm commented on VisualizationLib/SimplePyplot/SimplePyplot.py:
```diff
@@ -0,0 +1,90 @@
+@date: 2018-Today
+@author: jasper.bathmann@ufz.de
+"""
+import numpy as np
```
numpy is not used

jbathmann replied:
removed

jbathmann commented on VisualizationLib/SimplePyplot/SimplePyplot.py:
```diff
@@ -0,0 +1,90 @@
+                                fc=colors.colors[i],
+                                edgecolors="k",
+                                label=tree_group.name)
+            leg = patch.Patch(color=colors.colors[i], label=tree_group.name)
```
fixed

jbathmann commented on VisualizationLib/SimplePyplot/SimplePyplot.py:
```diff
@@ -0,0 +1,90 @@
+            p = PatchCollection(patches_group,
+                                alpha=1,
+                                linewidths=1,
+                                fc=colors.colors[i],
```
thanks! seems to be an issue with windows. changed now

jbathmann commented:
> cool thing, but the labels in the figure are still wrong and I couldn't fix it...

Thanks for the review.

jbathmann commented:
I just added the inline documentation for all added visualization classes.
Additionally, I found a formatting error and replaced all occurrences of @VAR -> @param within the whole MANGA package.
